src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathRecommendationService.java

```diff
@@ -506,51 +506,46 @@ private static int selectExercisesWithDifficulty(Map<DifficultyLevel, Set<Exerci
      * @return the average confidence of all prior competencies
      */
     private static double computeCombinedPriorConfidence(Competency competency, RecommendationState state) {
-        return state.priorCompetencies.get(competency.getId()).stream().map(state.competencyIdMap::get).map(c -> c.getUserProgress().stream().findFirst())
-                .mapToDouble(competencyProgress -> {
-                    if (competencyProgress.isEmpty()) {
-                        return 0;
-                    }
-                    return competencyProgress.get().getConfidence();
-                }).sorted().average().orElse(100);
+        return state.priorCompetencies.get(competency.getId()).stream().map(state.competencyIdMap::get).flatMap(c -> c.getUserProgress().stream())
+                .mapToDouble(CompetencyProgress::getConfidence).sorted().average().orElse(1);
     }
 
     /**
-     * Predicts the number of exercises required to master the given competency based on prior performance and current progress.
+     * Predicts the additionally needed exercise points required to master the given competency based on prior performance and current progress.
      * <p>
      * The following formulas are used predict the number of exercises required to master the competency:
      * <ul>
      * <li>Mastery >= MasteryThreshold</li>
-     * <li>Mastery = 1/3 * Progress + 2/3 * Confidence</li>
-     * <li>Progress = (#CompletedLearningObjects + #ExercisesRequiredToMaster) / #LearningObjects</li>
-     * <li>Confidence = (Sum of LatestScores + #ExercisesRequiredToMaster * avg. prior Confidence) / (#LatestScores + #ExercisesRequiredToMaster)</li>
+     * <li>Mastery = Progress * Confidence</li>
+     * <li>Progress = (#Exercises / # LearningObjects) * (AchievedPoints / TotalPoints) + #LectureUnits / #LearningObjects</li>
+     * <li>Confidence ≈ 0.9 * weightedConfidence</li>
+     * <li>RequiredPoints = AchievedPoints - CurrentScore</li>
```

Comment:
I am not sure if it is a good idea to specify it that detailed in the javadoc, I think that there is a high chance of the javadoc not being aligned with the actual code in the future.
It might be a better approach to link the related variables via javadoc?

Reply:
Especially the Progress calculation can be a bit hard to directly see from the method, so having a clear formula here is beneficial in my opinion. Additionally with the maintainer approvals I hope that we catch such problems in future PRs. But I also added links to the relevant methods